R/log_shiny_input_changes.R

```diff
@@ -0,0 +1,71 @@
+#' Auto logging input changes in Shiny app
+#'
+#' This is to be called in the \code{server} section of the Shiny app.
+#'
+#' Function having very similar behavior as [logger::log_shiny_input_changes()] but adjusted for `teal` needs.
+#'
+#' @param input passed from Shiny \code{server}
+#' @param excluded_inputs character vector of input names to exclude from logging
+#' @param excluded_patterns character of length one including a grep pattern of names to be excluded from logging
+#' @param namespace the name of the namespace
+#' @examples
+#' \dontrun{
+#' library(shiny)
+#'
+#' ui <- bootstrapPage(
+#'   numericInput("mean1", "mean1", 0),
+#'   numericInput("mean2", "mean2", 0),
+#'   numericInput("sd", "sd", 1),
+#'   textInput("title", "title", "title"),
+#'   textInput("foo", "This is not used at all, still gets logged", "foo"),
+#'   passwordInput("password", "Password not to be logged", "secret"),
+#'   plotOutput("plot")
+#' )
+#'
+#' server <- function(input, output) {
+#'   log_shiny_input_changes(input, excluded_inputs = "password", exclude_patterns = "mean")
+#'
+#'   output$plot <- renderPlot({
+#'     hist(rnorm(1e3, input$mean, input$sd), main = input$title)
+#'   })
+#' }
+#'
+#' shinyApp(ui = ui, server = server)
+#' }
+#' @export
+log_shiny_input_changes <- function(
+    input,
+    namespace = NA_character_,
+    excluded_inputs = character(),
+    excluded_patterns = "_width$") {
```

Comment:
- if this is a single string then singular: `excluded_patterns` -> `excluded_pattern` (+ docs update)

Reply:
duly noted! corrected in here 9a52a5194bd6b8dc8c5d84e99bbc1e66c507f843

Reply:
please also fix example code

Reply:
there was a typo in the example and the GSUB didn't change it... thanks for noticing, changing in a moment

Reply:
https://github.com/insightsengineering/teal.logger/pull/84/commits/049e630a8fb9948e5be86b5f575b3a691e6810d5